app/src/main/scala/org/alephium/app/ServerUtils.scala

```diff
@@ -64,6 +65,25 @@ class ServerUtils(implicit
     }
   }
 
+  def averageHashRate(blockFlow: BlockFlow, fromTs: TimeStamp, toTs: TimeStamp)(implicit
+      groupConfig: GroupConfig
+  ): Try[HashRateResponse] = {
+    blockFlow.getHeightedBlocks(fromTs, toTs) match {
+      case Right(blocks) =>
+        val hashCount = blocks.fold(BigInt(0)) { case (acc, entries) =>
+          entries.fold(acc) { case (hashCount, entry) =>
+            val target   = entry._1.target
+            val hashDone = Target.maxBigInt.divide(target.value)
+            hashCount + hashDone
+          }
+        }
+        val timeSpan = toTs.deltaUnsafe(fromTs)
+        val hashrate = (hashCount * 1000 * groupConfig.chainNum) / timeSpan.millis
```

Comment:
We could use this function to calculate hashrate: https://github.com/alephium/alephium/blob/master/flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala#L220-L221

Reply:
As discussed, we will not use this method